core/src/staked_validators_cache.rs

```diff
@@ -45,6 +46,12 @@ pub struct StakedValidatorsCache {
 
     /// Whether to include the running validator's socket address in cache entries
     include_self: bool,
+
+    /// Optional override for Alpenglow port, used for testing purposes
+    alpenglow_port_override: Option<AlpenglowPortOverride>,
+
+    /// timestamp of the last alpenglow port override
+    alpenglow_port_override_timestamp: Instant,
```

Comment:
you can tuck this in the `AlpenglowPortOverride` type now

Reply:
I'm okay with what you have here.

If we want to tuck it inside `AlpenglowPortOverride`, then we don't need two timestamps (see other comment below)